rust/parquet/src/data_type.rs

```diff
@@ -327,7 +422,12 @@ pub trait AsBytes {
 pub trait SliceAsBytes: Sized {
     /// Returns slice of bytes for a slice of this data type.
     fn slice_as_bytes(self_: &[Self]) -> &[u8];
-    fn slice_as_bytes_mut(self_: &mut [Self]) -> &mut [u8];
+    /// Return the internal representation as a mutable slice
+    ///
+    /// # Safety
+    /// If modified you are _required_ to ensure the internal representation
+    /// is valid and correct for the actual raw data
+    unsafe fn slice_as_bytes_mut(self_: &mut [Self]) -> &mut [u8];
```

Comment:
Notes to other reviewers -- this isn't any new unsafe that was added -- I think it was just hoised into a trait (previously the code inside the impl was marked as unsafe)

Reply:
yes, I made this more obviously unsafe given its sneaky type breaking nature, I wanted to highlight a bit more obviously what this was doing